Core/src/main/java/com/plotsquared/core/util/PlayerManager.java

```diff
@@ -211,6 +216,27 @@ public static void getUUIDsFromString(
         return StaticCaption.of(name);
     }
 
+    public CompletableFuture<Caption> getUsernameCaption(@Nullable UUID uuid) {
+        if (uuid == null) {
+            return CompletableFuture.completedFuture(TranslatableCaption.of("info.none"));
+        }
+        if (uuid.equals(DBFunc.EVERYONE)) {
+            return CompletableFuture.completedFuture(TranslatableCaption.of("info.everyone"));
+        }
+        if (uuid.equals(DBFunc.SERVER)) {
+            return CompletableFuture.completedFuture(TranslatableCaption.of("info.server"));
+        }
+        if (playerMap.containsKey(uuid)) {
+            return CompletableFuture.completedFuture(StaticCaption.of(playerMap.get(uuid).getName()));
+        }
```

Comment:
This seems to be a general problem in this class, but this is subject to a race condition. It would be safer to call get directly and do a null check then.

As a follow-up to this PR, it might be a good idea to generally a) either specify which threads are allowed to interact with which methods safely in this class, and/or b) guard access to the playerMap by RW locking or use a concurrent map.